### ISSUE TITLE
seek for help

### ISSUE THREAD

**Issue by lulupear** [hinj/hInjector#1]
I am a researcher in Beijing Jiaotong University. I performed hInjector at my computer without patching. Xen may crashed, in this case I suppose, but it dosen't. The experimental environment is built as exactly as you stated in the paper except code of xen-4.4.1 is modified for vulnerability CVE-2012-5513. I do not know why. Should I try xen-4.1.0?  I really want you to give me some advises. I really expect and appreciate for your help.

**Comment by CSendner** [comment]
Hey! Without patching the xen-source, it won’t work. You need to alter the shared_info struct in order to communicate with the also build-in filter. Otherwise every hypercall will be intercepted and none will be executed. Greetings, Christoph

**Comment by lulupear** [comment]
Thank you for your response. I didn't patch because i thought patching will keep Xen from crashing(the component "filter" will works if i patched and "filter" keeps Xen from crashing"), doesn't it?

**Comment by CSendner** [comment]
Hey, yes, you’re right. We didn’t want Xen to crash, so we can run more tests without restarting. You can disable the filter by changing the „offset“ in your hinjLKM.c at line 517 or just comment it.

**Comment by lulupear** [comment]
OK, thanks!!  I will try as you said, thanks again!

**Comment by lulupear** [comment]
Hey, I am sorry to bother you again, i tried but failed again. I run  hInjector.py in Dom0 other than a guest virtual machine, is that ok?(value of domid in config has been changed to "0" )  Is there any other place i shoud alter? 
expect your response T.T

**Comment by CSendner** [comment]
To get some things straight:

your Xen is working fine. In order to crash it, you built an xml-file and tried to run it with the hinjector. But it won’t crash. Do you get some errors via dmsg? Is everything compiled? (sender and LKM) Did you alter your arch_shared_info-struct within the kernel/system-headers?

https://github.com/hinj/hInjector/commit/1e3d9037799955a621c73441d81f3fc434ba4728 https://github.com/hinj/hInjector/commit/1e3d9037799955a621c73441d81f3fc434ba4728 <— did you read these instructions?

**Comment by lulupear** [comment]
yes, I have read those instructions before experiment and altered arch_shared_info-struct within the kernel/system-headers, I will check everything again and try to get some message via dmsg. Thanks for advice. :-D  ...in addition, I performed hInjector in dom0, is that ok? because i have problem creating guest virtual machine. Greetings, LuLi

**Comment by CSendner** [comment]
Yeah, it should be okay. What kind of problems creating a guest do you have?

**Comment by lulupear** [comment]
Kinds of syntax errors turned up when i compiled libvirt using “make” and we didn‘t work it out over a week.

**Comment by CSendner** [comment]
Any progress compiling xen?

**Comment by lulupear** [comment]
Sorry that i just find this message. I have suspended the project for making time to get a job and just return to previous work these days. And i did trigger the vulnerability once but i can't succeed again. Obviously, i have not  really figure it out.

**Comment by allewwaly** [comment]
After comments  line 517 at hinjLKM.c and executes the example.xml, my system going to reboot. And after that, no VMs can be created, it just stucks at parsing config file, even after I recompile xen.
What does example.xml do to my system and how to fix that?

**Comment by Preetixen** [comment]
HI,
sorry but could you tell me if you tried HInjector
Can HInjector be patched with xen 4.6 and does it have some vulnerabiility corresponding to that version?
